packages/testcontainers/src/chains/reg_test_container/masternode.ts

```diff
@@ -173,4 +181,37 @@ export class MasterNodeRegTestContainer extends RegTestContainer {
     const getaddressinfo = await this.call('getaddressinfo', [address])
     return { address, privKey, pubKey: getaddressinfo.pubkey }
   }
+
+  async sync (peers: MasterNodeRegTestContainer[]): Promise<void> {
+    // Note(canonbrother): make block count diff before sync
+    await this.generate(1)
+
+    for (let i = 0; i < peers.length; i += 1) {
+      const peer = peers[i]
+      await this.addNode(peer.ip)
+    }
+
+    await this.waitForSync([this, ...peers])
+  }
+
+  async waitForSync (containers: MasterNodeRegTestContainer[], timeout: number = 60000): Promise<void> {
+    return await this.waitForCondition(async () => {
+      const bestBlockHashes = await Promise.all(containers.map(async (c: MasterNodeRegTestContainer) => {
+        try {
+          // Note(canonbrother): node may froze here, do restart -> addNode -> waitForSync again
+          return await c.call('getbestblockhash')
+        } catch (err) {
+          await c.restart()
+          await this.addNode(c.ip)
+          await this.waitForSync(containers)
+        }
```

Comment:
hmm.. i think it can be optimized here
the bNode will be froze while link (eg: aNode -- link --> bNode)